django/core/cache/backends/base.py

```diff
@@ -161,17 +161,17 @@ def get_or_set(self, key, default, timeout=DEFAULT_TIMEOUT, version=None):
 
         Return the value of the key stored or retrieved.
         """
-        val = self.get(key, version=version)
-        if val is None:
+        if self.has_key(key, version=version):
+            return self.get(key, version=version)
+        else:
```

Comment:
This approach is problematic, this will cause a double roudtrip to the cache server. A better approach would be to pass a _sentinel_ object to `get` that is different from `None` so you can differentiate between the two.